### PR TITLE
Add missing documentation for the CryptographicOperations class.

### PR DESCRIPTION
**Pull request by bartonjs**
cc @carlossanlop

### CODE REVIEW

File: xml/System.Security.Cryptography/CryptographicOperations.xml [hunk 1]
```diff
@@ -89,9 +89,19 @@ Fixed-time behavior is guaranteed in all other cases, including when `left` and
...
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="buffer">The buffer to fill with zeros.</param>
+        <summary>Fills the provided buffer with zeros.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method exists to future-proof against potential optimizations in the .NET runtime which could eliminate memory writes that are provably not followed by memory reads,
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography/CryptographicOperations.xml]: nit - small typo:
```suggestion
This method exists to future-proof against potential optimizations in the .NET runtime which could eliminate memory writes that are probably not followed by memory reads,
```

File: xml/System.Security.Cryptography/CryptographicOperations.xml [hunk 2]
```diff
@@ -89,9 +89,19 @@ Fixed-time behavior is guaranteed in all other cases, including when `left` and
...
+similar to optimizations made by C and C++ compilers.
+No such optimizations are planned at this time.
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography/CryptographicOperations.xml]: I think this sentence could confuse users. What does "at this time" mean? They would not know how long ago this text was added, or if it's out of date. Maybe we can reword this to be more specific?
```suggestion
No such optimizations are planned at this time.
```

File: xml/System.Security.Cryptography/CryptographicOperations.xml (hunk 1 above)
Comment (mairaw) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: I've added a small change to the suggestion above as well

which -> that

File: xml/System.Security.Cryptography/CryptographicOperations.xml (hunk 2 above)
Comment (mairaw) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: Sometimes we do say things as "as of late 2019."

e.g. https://github.com/dotnet/docs/search?q=%22as+of%22&unscoped_q=%22as+of%22
(conceptual docs have date published though so we can be a bit more vague.

**Comment by mairaw** [comment]
LGTM module the comments from @carlossanlop. So feel free to merge @carlossanlop when it's ready.

File: xml/System.Security.Cryptography/CryptographicOperations.xml (hunk 1 above)
Comment (bartonjs) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: I meant "provably" (able to be proven), not "probably" (may or may not happen)

File: xml/System.Security.Cryptography/CryptographicOperations.xml (hunk 2 above)
Comment (bartonjs) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: > Maybe we can reword this to be more specific?

It's an open-ended defense.  The point of the method is "maybe one day we'd do this, so you should declare your intent now and be safe in the indefinite-and-maybe-never-going-to-happen future".  I'm happy to accept suggestions, but leaving off the trailing sentence I feel implies we're planning them... but we're not, it's just theoretical.

Comment (carlossanlop) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: No, let's keep the trailing sentence. I can't think of a better way to reword this, and it's no big deal. I'll approve the PR, and if @mairaw has a suggestion to get it reworded, let's consider it.

Comment (gewarren) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: While I think this is a little different because it's talking about what's *not* planned for the future, there is [this guidance](https://review.docs.microsoft.com/en-us/help/contribute/contribute-get-started-channel-guidance?branch=master#content-that-does-not-belong-in-a-docsmicrosoftcom-article-or-repository) in our contributor's guide:

Future product plans: Do not publish statements about future product plans in technical documentation. Technical documentation should describe only what is possible in the released product. Why?

It’s not actionable: Technical content should be about what customers can and should do today, not what they might be able to do some day.
It erodes trust: If a writer says a feature is coming soon and it is delayed or canceled, the statement risks eroding trust for all our technical documentation.
Looks like marketing: Talking about the future looks more like marketing material (aspirational rather than reality).

File: xml/System.Security.Cryptography/CryptographicOperations.xml [hunk 3]
```diff
@@ -89,9 +89,19 @@ Fixed-time behavior is guaranteed in all other cases, including when `left` and
...
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="buffer">The buffer to fill with zeros.</param>
+        <summary>Fills the provided buffer with zeros.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method exists to future-proof against potential optimizations in the .NET runtime that could eliminate memory writes that are provably not followed by memory reads,
```
Comment (gewarren) [inline xml/System.Security.Cryptography/CryptographicOperations.xml]:
```suggestion
This method exists to future-proof against potential optimizations in the .NET runtime that could eliminate memory writes that aren't followed by memory reads.
```

File: xml/System.Security.Cryptography/CryptographicOperations.xml [hunk 4]
```diff
@@ -89,9 +89,19 @@ Fixed-time behavior is guaranteed in all other cases, including when `left` and
...
+similar to optimizations made by C and C++ compilers.
```
Comment (gewarren) [inline xml/System.Security.Cryptography/CryptographicOperations.xml]:
```suggestion
While the C and C++ compilers have similar optimizations, no such optimizations are planned at this time for .NET.
```

File: xml/System.Security.Cryptography/CryptographicOperations.xml [hunk 5]
```diff
@@ -89,9 +89,19 @@ Fixed-time behavior is guaranteed in all other cases, including when `left` and
...
+No such optimizations are planned at this time.
```
Comment (gewarren) [inline xml/System.Security.Cryptography/CryptographicOperations.xml]:
```suggestion
```

File: xml/System.Security.Cryptography/CryptographicOperations.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: Oh interesting. I didn't know that word existed. Thank you, native English speaker 😄 . Feel free to apply @mairaw 's suggestion alone if you want, @bartonjs.

File: xml/System.Security.Cryptography/CryptographicOperations.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Security.Cryptography/CryptographicOperations.xml]: ^ @bartonjs Genevieve found good guidance and she has a point, in case you may want to reconsider removing the sentence about future plans.
